utils/estimated_histogram.hh

```diff
@@ -55,70 +55,95 @@
 namespace utils {
 
 /**
- * This is a pseudo floating point implementation of an estimated histogram.
- * When entering a value:
- * All values lower than the MIN will be included in the first bucket.
- * All values higher than MAX will be included the last bucket that serves as the
- * infinity bucket.
+ * This is a pseudo-exponential implementation of an estimated histogram.
  *
- * buckets are distributed as pseudo floating point:
- * The range [MIN, MAX) is split into log2 ranges.
- *     ranges = log2(max/min)
- * Each of that ranges is split according to the number of buckets:
- *     resolution = (NUM_BUCKETS - 1)/ranges
+ * An exponential-histogram with coefficient 'coef', is a histogram where for bucket 'i'
+ * the lower limit is coef^i and the higher limit is coef^(i+1).
  *
- * For example, if the MIN value is 128, the MAX is 1024 and the number of buckets is 13:
+ * A pseudo-exponential is similar but the bucket limits are an approximation.
+ *
+ * The approx_exponential_histogram uses Half-precision floating-point (or pseudo-floating-point)
```

Comment:
https://en.wikipedia.org/wiki/Half-precision_floating-point_format

Reply:
It means that a bucket representation has two parts upper bits are the exponential part, lower bits are he linear part

Reply:
1. Again, yes, there is strong *analogy* or *similarity* between what you do here to here to how floating-point works - with exponent bits (exponentials) and mantissa bits (linear). You can definitely say that and it will indeed be nice to have this explained.
2. You do not *use* floating point, so you shouldn't use the verb "uses".
3. I don't see any relevant to "half precision floating point" which is indeed half of single-precision, i.e., two bytes. It's not any more similar to what you do than single precision or double precision floating point or seven-and-a-half-precision ;-) Unless I'm missing something?

Reply:
Thanks,
I was hoping that specifying a specific format would make the explanation clearer.
If it doesn't I'll re-word it